atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraTimestampBackupRunner.java

```diff
@@ -43,8 +44,10 @@
 
     private final CassandraKeyValueService cassandraKeyValueService;
 
-    public CassandraTimestampBackupRunner(CassandraKeyValueService cassandraKeyValueService) {
-        this.cassandraKeyValueService = cassandraKeyValueService;
+    public CassandraTimestampBackupRunner(KeyValueService keyValueService) {
+        Preconditions.checkArgument(keyValueService instanceof CassandraKeyValueService,
```

Comment:
In the end I added a bunch of unit tests for `CassandraTimestampStoreInvalidator` (where this now lives)